src/pages/profile/index.ts

```diff
@@ -29,6 +31,32 @@ export default class ProfilePage
     m.redraw()
   }
 
+  async sendMessage(contact: Contact, message: IMessage) {
+    const internalMessage = {
+      sender: agent.profile.label,
+      receiver: contact.label || contact.did,
+      timestamp: new Date(),
+      type: message.type,
+      content: "",
+      raw: message,
+    }
+    await agent.sendMessage(contact, message)
+    internalMessage.raw.from = agent.profile.did
+    ContactService.addMessage(contact.did, internalMessage)
+  }
+
+  async sendProfile(contact: Contact) {
+    const message = {
+      type: "https://didcomm.org/user-profile/1.0/profile",
+      body: {
+        profile: {
+          displayName: agent.profile.label,
+        }
+      }
+    }
+    await this.sendMessage(contact, message as IMessage)
+  }
```

Comment:
I think we should move the protocol details into `agent.ts` -- so maybe that would mean that we add a `sendProfile` method to the `Agent` class.